File: Simplex.Core/Actions/IActionBinder.cs
using System;
using System.Collections.Generic;
using System.Reactive;
using System.Text;

namespace Tmon.Simplex.Actions
{
    public interface IActionBinder { }

    public interface IActionBinder<out TAction, TResult> : IActionBinder
        where TAction : IAction<TResult>
    {
        TAction Action { get; }
    }

    public interface IActionBinder<out TAction, TParam, TResult> : IActionBinder<TAction, TResult>
        where TAction : IAction<TParam, TResult>
    {
    }
}

File: Simplex.kotlin/src/main/kotlin/kr/co/tmon/simplex/actions/IActionBinder.kt
package kr.co.tmon.simplex.actions

/**
 * 액션 래핑 인터페이스
 * @param <TAction> 액션 타입
 * @param <TParam> 액션의 매개변수 타입
 * @param <TResult> 액션의 결과값 반환 타입
*/
interface IActionBinder<TAction : IAction<TParam, TResult>, TParam, TResult> {
	val action: TAction 
}
